airflow/sensors/base_sensor_operator.py

```diff
@@ -159,6 +160,16 @@ def _get_next_poke_interval(self, started_at, try_number):
         else:
             return self.poke_interval
 
+    def prepare_for_execution(self) -> BaseOperator:
```

Comment:
```suggestion
    def prepare_for_execution(self) -> "BaseSensorOperator":
```
Lovely! Now we have this method it's indeed easier to add some custom "pre_execution" logic.

Reply:
Incompatible return value :(